shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_property_object_impl.cpp

```diff
@@ -22,38 +22,55 @@ using namespace opcua;
 template <class Impl>
 ErrCode TmsClientPropertyObjectBaseImpl<Impl>::setPropertyValueInternal(IString* propertyName, IBaseObject* value, bool protectedWrite)
 {
-    return daqTry(
+    if (propertyName == nullptr)
+    {
+        LOG_W("Failed to set value for property with nullptr name on OpcUA client property object");
+        return OPENDAQ_IGNORED;
+    }
+    auto propertyNamePtr = StringPtr::Borrow(propertyName);
+
+    StringPtr lastProccessDescription = "";
+    ErrCode errCode = daqTry(
         [&]()
         {
-            if (const auto& it = introspectionVariableIdMap.find((StringPtr) propertyName); it != introspectionVariableIdMap.cend())
+            if (const auto& it = introspectionVariableIdMap.find(propertyNamePtr); it != introspectionVariableIdMap.cend())
             {
                 if (protectedWrite)
                 {
+                    lastProccessDescription = "Checking exisiting property is read-only";
                     PropertyPtr prop;
                     checkErrorInfo(getProperty(propertyName, &prop));
                     const bool readOnly = prop.getReadOnly();
                     if (readOnly)
                         return OPENDAQ_SUCCESS;
                 }
 
+                lastProccessDescription = "Writting property value";
                 const auto variant = VariantConverter<IBaseObject>::ToVariant(value, nullptr, daqContext);
                 client->writeValue(it->second, variant);
                 return OPENDAQ_SUCCESS;
             }
 
-            if (const auto& it = referenceVariableIdMap.find((StringPtr) propertyName); it != referenceVariableIdMap.cend())
+            if (const auto& it = referenceVariableIdMap.find(propertyNamePtr); it != referenceVariableIdMap.cend())
             {
+                lastProccessDescription = "Setting property value";
                 const auto refProp = this->objPtr.getProperty(propertyName).getReferencedProperty();
                 return setPropertyValue(refProp.getName(), value);
             }
 
-            if (const auto& it = objectTypeIdMap.find((StringPtr) propertyName); it != objectTypeIdMap.cend())
+            if (const auto& it = objectTypeIdMap.find((propertyNamePtr)); it != objectTypeIdMap.cend())
             {
-                return this->makeErrorInfo(OPENDAQ_ERR_NOTIMPLEMENTED, "Object type properties cannot be set over OPC UA");
+                lastProccessDescription = "Object type properties cannot be set over OpcUA";
+                return OPENDAQ_ERR_NOTIMPLEMENTED;
             }
-
+            lastProccessDescription = "Property not found";
             return OPENDAQ_ERR_NOTFOUND;
         });
+    if (OPENDAQ_FAILED(errCode))
```

Comment:
I'd return an error for `OPENDAQ_NOT_FOUND` here, similarly as done in the core.